eventmesh-connectors/eventmesh-connector-http/src/main/java/org/apache/eventmesh/connector/http/sink/handle/CommonHttpSinkHandler.java

```diff
@@ -0,0 +1,197 @@
+            .setKeepAliveTimeout(this.connectorConfig.getKeepAliveTimeout() / 1000)
+            .setIdleTimeout(this.connectorConfig.getIdleTimeout())
+            .setIdleTimeoutUnit(TimeUnit.MILLISECONDS)
+            .setConnectTimeout(this.connectorConfig.getConnectionTimeout())
+            .setMaxPoolSize(this.connectorConfig.getMaxConnectionPoolSize());
+        this.webClient = WebClient.create(vertx, options);
+    }
+
+
+    @Override
+    public void handle(ConnectRecord record) {
+        // create headers
+        MultiMap headers = HttpHeaders.headers()
+            .set(HttpHeaderNames.ACCEPT, "application/json; charset=utf-8");
+
+        // convert ConnectRecord to HttpConnectRecord
+        HttpConnectRecord httpConnectRecord = convertToHttpConnectRecord(record);
+
+        // send the request
+        this.webClient.post(this.connectorConfig.getPath())
+            .putHeaders(headers)
+            .sendJson(httpConnectRecord)
+            .onSuccess(res -> {
+                Long timestamp = record.getTimestamp();
+                Map<String, ?> offset = record.getPosition().getOffset().getOffset();
+                log.info("Request sent successfully. Record: timestamp={}, offset={}", timestamp, offset);
+                // Determine whether the status code is 200
+                if (res.statusCode() == HttpResponseStatus.OK.code()) {
+                    // store the received data, when webhook is enabled
+                    if (this.connectorConfig.isWebhook()) {
+                        String dataStr = res.body().toString();
+                        if (dataStr.isEmpty()) {
+                            log.warn("Received data is empty.");
+                            return;
+                        }
+                        JSONObject receivedData = JSON.parseObject(dataStr);
+                        if (receivedDataQueue.size() == Integer.MAX_VALUE) {
+                            // if the queue is full, remove the oldest element
+                            JSONObject removedData = receivedDataQueue.poll();
+                            log.info("The queue is full, remove the oldest element: {}", removedData);
+                        }
+                        boolean b = receivedDataQueue.offer(receivedData);
+                        if (b) {
+                            log.info("Successfully put the received data into the queue: {}", receivedData);
+                        } else {
+                            log.error("Failed to put the received data into the queue: {}", receivedData);
+                        }
+                    }
+                } else {
+                    log.error("Unexpected response received. Record: timestamp={}, offset={}. Response: code={} header={}, body={}",
+                        timestamp,
+                        offset,
+                        res.statusCode(),
+                        res.headers(),
+                        res.body().toString()
```

Comment:
In fact, the sink connector does not expect any "expected data" from the callbackUrl, because it never tries to understand the data it gets.

Since the logging here is judged by the HTTP status code, I think it's sufficient to log it as it is under warning logging level (any error level log is considered as an application error, the non-2XX response is not the connector's fault): "Received request with response code 502. Record: timestamp={}, offset={}". (and print response string under debug level)

Second, all non-2XX responses are dropped here. They should be queued and retrieved when `/export` is called.

By the way, IMO, even putting these parameters on the same line does not affect readability.

Reply:
Kindly notice that I haven't seen any reply in this conversation.

Reply:
Modified but not yet submitted.

Reply:
Resolved at https://github.com/apache/eventmesh/pull/4837/commits/fe2b7327b744fcf0f3ea3576c892c551ed1d6fb4